src/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,25 @@
+import {Image} from 'react-native-compressor';
+import {composeImageMeta} from '../utils/string/file';
+import {uploadPhoto} from '../service/file';
+
+const useUploadImage = () => {
+  const uploadPhotoImage = async (pathImg: string) => {
+    try {
+      const compressionResult = await Image.compress(pathImg, {
+        compressionMethod: 'auto'
+      });
+      const asset = new FormData();
+      asset.append('photo', composeImageMeta(compressionResult));
+      const responseUpload = await uploadPhoto(asset);
+      return responseUpload;
+    } catch (e) {
+      return undefined;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Returning `undefined` when an error occurs might make it difficult for the calling code to handle errors properly. It would be better to throw the error again after catching it, or at least return some kind of error object.

```diff
-      return undefined;
+      throw e;
```

25:
As mentioned above, since this is not a hook, it should not be exported as default. Instead, export it as a named export.

```diff
- export default useUploadImage;
+ export { createUploadImageFunction };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->